### ISSUE TITLE
DNS Zone supports creating Record, but not adding Record

### ISSUE THREAD

**Issue by quartzmo** [GoogleCloudPlatform/google-cloud-ruby#1023]
The docs for [`Zone#record`](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-dns/lib/google/cloud/dns/zone.rb#L307-L323) state:

> Creates a new, unsaved Record that can be added to a Zone.

And the example shows this as well:

```ruby
dns = Google::Cloud::Dns.new
zone = dns.zone "example-com"
record = zone.record "example.com.", "A", 86400, ["1.2.3.4"]
zone.add record
```

However, while working on testing examples for #226, I discovered that `Zone#add` does not accept a record:

```sh
  1) Error:
Google::Cloud::Dns::Zone#record#test_0001_:
ArgumentError: wrong number of arguments (given 1, expected 4)
    /Users/quartzmo/code/google/codez/gcloud-ruby/google-cloud-dns/lib/google/cloud/dns/zone.rb:534:in `add'
    /Users/quartzmo/code/google/codez/yard-doctest/lib/yard/doctest/example.rb:81:in `context'
    /Users/quartzmo/code/google/codez/gcloud-ruby/google-cloud-dns/lib/google/cloud/dns/zone.rb:320
```

I cannot find any way to save the transient `Record` instance returned by `Zone#record`. Is there one that I'm missing?

**Comment by blowmage** [comment]
I believe the code example is wrong. It should be the following:


```ruby
dns = Google::Cloud::Dns.new
zone = dns.zone "example-com"
zone.add "example.com.", "A", 86400, ["1.2.3.4"]
```

Only `update` takes Record objects. The `add` method take the individual record values, same as `record`. This is all documented in the method parameters. Only the code examples is wrong.

**Comment by quartzmo** [comment]
Thanks, I'll fix the example in my PR for DNS example testing.

**Comment by quartzmo** [comment]
Closed by PR #1026.
